### PR TITLE
[1840] Initial Map + Tiles Commit

### PR DESCRIPTION
**Pull request by Zwergenpunk**
Added Map and Tiles.

### CODE REVIEW

File: assets/app/lib/hex.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ module Hex
       black: '#000000',
       red: '#ec232a',
       blue: '#35A7FF',
+      purple: '#A79ECD'
```
Comment (yzemaze) [inline assets/app/lib/hex.rb:33]: cf. https://github.com/tobymao/18xx/pull/4594#discussion_r597455761

File: public/icons/1840/yellow_hex.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="32" height="32"  stroke="#000" fill="#fde900" xmlns:v="https://vecta.io/nano"><path stroke-width="2" d="M2.5 13.049L16 3.5l13.5 9.549L24.343 28.5H7.657L2.5 13.049z"/><path d="M1 291.764l149-.382 149 .382-56.913.618H57.913L1 291.764z"/></svg>
```
Comment (yzemaze) [inline public/icons/1840/yellow_hex.svg:1]: `stroke="#000"`
… is default. Remove.

`xmlns:v="https://vecta.io/nano"`
Remove.
(… from all svg files)

File: lib/engine/config/tile.rb [hunk 3]
```diff
@@ -487,7 +496,9 @@ module Tile
         '469' => 'path=a:0,b:1,track:narrow',
       }.freeze
 
-      COLORS = %i[white yellow green brown gray blue greenbrown browngray none red].freeze
+      PURPLE = {}.freeze
```
Comment (yzemaze) [inline lib/engine/config/tile.rb:499]: Intention behind this?

Comment (Zwergenpunk) [reply lib/engine/config/tile.rb:499]: without changing either the COLORS or adding PURPLE (or both, did not try just one actually), I recevied an error

File: lib/engine/game/g_1840/entities.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G1840
+      module Entities
+        MARKET = [
```
Comment (tobymao) [inline lib/engine/game/g_1840/entities.rb:7]: run this file through an indenter.

File: lib/engine/game/g_1840/entities.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            %w[40],
+             ].freeze
+
+        PHASES = [
+            {
+              name: 'a',
+              train_limit: 3,
+              tiles: [:yellow],
```
Comment (tobymao) [inline lib/engine/game/g_1840/entities.rb:15]: can't you do like %i[yellow]

Comment (Zwergenpunk) [reply lib/engine/game/g_1840/entities.rb:15]: this file is just there that I can test the map

**Pull request merged**